client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -29,16 +51,87 @@ const getIframeUrl = ( siteAdminUrl, postId, postType ) => {
 };
 
 class CalypsoifyIframe extends Component {
+	state = {
+		isMediaModalVisible: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.iframe = null;
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'message', this.onMessage, false );
```

Comment:
see the security concerns below.

we may want to use a [MessageChannel](https://developer.mozilla.org/en-US/docs/Web/API/MessageChannel) to communicate with the `iframe` instead so that we can bypass the security element of determining if our message came from the iframe we expected it to.

```js
// adapted from the MDN guide
const channel = new MessageChannel();

iframe.addEventListener("load", {
	channel.port1.onmessage = this.messageFromIFrame;
	iframe.contentWindow.postMessage( 'establish-communication', '*', [channel.port2] );
} );
```